antenna/heartbeat.py

```diff
@@ -0,0 +1,92 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""The Antenna app has a set of things that need to run at a specified
+interval. We call this interval "the heartbeat". Depending on how you have
+Antenna configured, different things could be running every heartbeat.
+
+This file provides infrastructure for creating a HeartbeatManager, registering
+functions that should run every heartbeat and running functions in a way that
+captures unhandled exceptions but doesn't otherwise disturb the heartbeat.
+
+"""
+
+import logging
+
+from everett.component import ConfigOptions, RequiredConfigMixin
+import gevent
+
+from antenna.sentry import capture_unhandled_exceptions
+
+
+logger = logging.getLogger(__name__)
+
+
+class HeartbeatManager(RequiredConfigMixin):
+    """Heartbeat manager
+
+    This holds heartbeat state and the methods used to start, stop and run the
+    heartbeat.
+
+    """
+
+    required_config = ConfigOptions()
+
+    # Interval between heartbeats
+    heartbeat_interval = 10
```

Comment:
Why 10? Seemed like a good number that's long enough that batching is helpful but not so long that it's annoying.